data_structures/stacks/stack.py

```diff
@@ -66,17 +66,18 @@ def peek(self) -> T:
 
     def is_empty(self) -> bool:
         """Check if a stack is empty."""
-        return not bool(self.stack)
+        return (not self.stack)
 
     def is_full(self) -> bool:
+        """Check if a stack is full."""
         return self.size() == self.limit
 
     def size(self) -> int:
```

Comment:
Please change to
```suggestion
    def __len__(self) -> int:
```
https://docs.python.org/3/reference/datamodel.html#object.__len__ and then change all calls to `.size()`

Reply:
Message to the bot: I didn't want to change the actual code that much, but this isn't really a bad idea, just thought I'd keep it the way it was

Reply:
Not a bot.